exercicios/para-casa/entregas/Geice-SousaPinho/SavingAccount.js

```diff
@@ -0,0 +1,135 @@
+const { Bank } = require("./Bank");
+const { Client } = require("./Client");
+const { BankAccount } = require("./BankAccount");
+const { CurrentAccount } = require("./CurrentAccount");
+
+class SavingAccount extends BankAccount {
+  #qtdWithdrawal = 0;
+  #MAX_OF_WITHDRAWAL = 2;
+  #withdrawalTax = 0.03;
+  incomeRate;
+  incomeDay;
+
+  constructor(client, bank, account, agency, incomeDay, incomeRate) {
+    super(client, bank, account, agency);
+    this.incomeDay = incomeDay;
+    this.incomeRate = incomeRate;
+  }
+
+  cashWithdrawal(amount) {
+    console.log(
+      `**************\nOlá ${this.client.name.toUpperCase()}! Você tem ${
+        this.#MAX_OF_WITHDRAWAL
+      } retiradas gratuitas por mês. \n Retiradas realizadas: ${
+        this.#qtdWithdrawal
+      }.\n**************`
+    );
+
+    // Caso o cliente NAO tenha saldo suficiente
+    if (this.balance < amount) {
+      console.log(
+        `O saque não pode ser realizado, pois não há saldo suficiente. \n Saldo atual: R$${this.balance}.`
+      );
+    } else {
+      // Caso o cliente ainda tenha retiradas gratuitas
+      if (this.#qtdWithdrawal < this.#MAX_OF_WITHDRAWAL) {
+        this.balance -= amount;
+        this.#qtdWithdrawal++;
+        console.log(
+          `Retirada no valor de R$${amount} realizada com sucesso.\n Seu saldo restante é R$ ${this.balance.toFixed(
+            2
+          )}.\n Retiradas realizadas: ${this.#qtdWithdrawal}`
+        );
+
+        return;
+      }
+
+      // Caso o cliente NAO tenha retiradas gratuitas
+      if (this.#qtdWithdrawal >= this.#MAX_OF_WITHDRAWAL) {
+        const amountAndTax = amount + amount * this.#withdrawalTax;
+
+        if (this.balance < amountAndTax) {
+          console.log(
+            `Você não possui saldo suficiente para realizar a operação nessa quantia.\n Saldo atual: R$${this.balance.toFixed(
+              2
+            )}.`
+          );
+
+          return;
+        }
+
+        this.balance -= amountAndTax;
+        this.#qtdWithdrawal++;
+        console.log(
+          `Operação realizada com sucesso.\n Saldo atual: R$${this.balance.toFixed(
+            2
+          )}\n Taxa de saque: R$${
+            amount * this.#withdrawalTax
+          }\n Retiradas realizadas: ${this.#qtdWithdrawal}.`
+        );
+      }
+    }
+  }
+
+  get qtdWithdrawal() {
+    return this.#qtdWithdrawal;
+  }
+
+  get withdrawalTax() {
+    return this.#withdrawalTax;
+  }
+
+  set withdrawalTax(newTax) {
+    return (this.#qtdWithdrawal = newTax);
+  }
+
+  generateIncome(currentDay) {
+    if (currentDay === this.incomeDay) {
+      this.balance += this.balance * 0.4;
```

Comment:
Aqui você poderia substituir a taxa fixa de 0.4 por uma variável que calcula a o rendimento. Acho que facilitaria para quando quiser realizar modificações futuras.

ex:
const income = this.balance * this.incomeRate;
      this.balance += income;